internal/generator/services.go

```diff
@@ -492,6 +492,11 @@ func composeToServiceValues(
 					}
 					// if the cronjob is inpod, or the cronjob has an inpod flag override
 					if inpod || (cronjob.InPod != nil && *cronjob.InPod) {
+						cmd := cronjob.Command
+						// Lagoon enforces that only a single instance of a cronjob can run at any one time.
+						// https://man7.org/linux/man-pages/man1/flock.1.html
+						// https://www.gnu.org/savannah-checkouts/gnu/bash/manual/bash.html#Shell-Parameter-Expansion
+						cronjob.Command = fmt.Sprintf("flock -n /tmp/cron.lock.$(echo \"%d %s\" | sha256sum | awk '{ print $1 }') -c '%s'", idx, cmd, cmd)
```

Comment:
I think this may be improved in a couple of ways:
1. calculating the sha256sum in Go will vastly simplify the shell command. it was only using awk etc. because it was written in Bash before.
2. The command will need to be [shell escaped](https://github.com/alessio/shellescape/blob/853065111c4c5878f9eb9033c262159ecaca0a03/shellescape.go#L23-L41) otherwise it can break if it contains a single quote.
3. not sure if you want to throw a warning or error on duplicate cronjob commands? they'll produce the same sha256sum and so it will be a race to see which starts first.

Reply:
Happy for you to fix as you think it needs to be

Reply:
We already check for duplicate cronjobs by name, but not command.